### ISSUE TITLE
generateStaticParams with next export: Build fails when generateStaticParams returns an empty array of params

### ISSUE THREAD

**Issue by NasserBvB** [vercel/next.js#71862]
### Link to the code that reproduces this issue

https://github.com/NasserBvB/generate-static-params-issue

### To Reproduce

While building a Next.js application with `output: "export"` in `next.config.js`, I encountered an error regarding the missing `generateStaticParams()` function, despite it being implemented and returning an empty list. This error specifically affects dynamic pages in the project.

Notably, the build succeeds as expected when `generateStaticParams()` returns at least one article parameter in the list. The issue only arises when it returns an empty list (`[]`).

### Steps to Reproduce

1. Create a new Next.js application using `next@14.2.8`.
2. Add a dynamic page at `articles/[article]/page.js`.
3. Implement `generateStaticParams` in `articles/[article]/page.js` and set it to return an empty list (`[]`).
4. Configure `output: "export"` in `next.config.js`.
5. Run the build command.

### Expected Behavior

The build should succeed without any issues, recognizing that `generateStaticParams()` is implemented and returning an empty list of article slugs.

### Actual Behavior

An error occurs during the build, specifically stating:

```
Error: Page "/articles/[article]" is missing "generateStaticParams()" so it cannot be used with "output: export" config.
```

However, when `generateStaticParams()` returns a list with at least one article slug, the build completes successfully as expected.

### Additional Context

- I’ve confirmed that `generateStaticParams()` exists in `articles/[article]/page.js`, and the issue persists only when it returns an empty list.

### Expected Behavior

The build should succeed without any issues, recognizing that `generateStaticParams()` is implemented and returning an empty list of article slugs.

### Actual Behavior

An error occurs during the build, specifically stating:

```
Error: Page "/articles/[article]" is missing "generateStaticParams()" so it cannot be used with "output: export" config.
```

However, when `generateStaticParams()` returns a list with at least one article slug, the build completes successfully as expected.

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.0.0: Tue Sep 24 23:36:26 PDT 2024; root:xnu-11215.1.12~1/RELEASE_ARM64_T8103
  Available memory (MB): 16384
  Available CPU cores: 8
Binaries:
  Node: 18.19.0
  npm: 10.2.3
  Yarn: 1.22.22
  pnpm: 9.12.0
Relevant Packages:
  next: 14.2.8 // An outdated version detected (latest is 15.0.1), upgrade is highly recommended!
  eslint-config-next: 14.2.8
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.5.4
Next.js Config:
  output: export
```

### Which area(s) are affected? (Select all that apply)

create-next-app, Output (export/standalone), Script (next/script)

### Which stage(s) are affected? (Select all that apply)

next build (local)

**Comment by Riley1101** [comment]
Hey @NasserBvB , I take a look at the issue. Looks like you will have to return `article` in the param object since your dynamic path is `/article/[article]/page.tsx`.

Take a look at here:   https://github.com/NasserBvB/generate-static-params-issue/blob/25021349b5e2430bdc72fab2c0764dff50084448/src/app/articles/%5Barticle%5D/page.tsx#L25C1-L27C7
```
export async function generateStaticParams() {
  const posts = await mockArticles();

  return posts.map((post) => ({
    slug: post.title,
  }));
}
```

**Comment by NasserBvB** [comment]
Yes you are right but even though you correct it you will still get the same thing

**Comment by samsan1212** [comment]
> Hey @NasserBvB , I take a look at the issue. Looks like you will have to return `article` in the param object since your dynamic path is `/article/[article]/page.tsx`.
> 
> Take a look at here: https://github.com/NasserBvB/generate-static-params-issue/blob/25021349b5e2430bdc72fab2c0764dff50084448/src/app/articles/%5Barticle%5D/page.tsx#L25C1-L27C7
> 
> ```
> export async function generateStaticParams() {
>   const posts = await mockArticles();
> 
>   return posts.map((post) => ({
>     slug: post.title, // <- slug should be article
>   }));
> }
> ```

Hi, the error still appears with this method

**Comment by marcoshernanz** [comment]
Thanks for the report. This is covered by #61213. The misleading error was fixed in #95969, and current releases now explain that `output: "export"` requires at least one generated route. Please continue the feature discussion in #61213.

**Comment by brynary** [comment]
This closed issue has been automatically locked because it had no new activity for 2 weeks. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
